src/Sentry/Internal/IFileSystem.cs

```diff
@@ -8,14 +15,21 @@ internal interface IFileSystem
     IEnumerable<string> EnumerateFiles(string path);
     IEnumerable<string> EnumerateFiles(string path, string searchPattern);
     IEnumerable<string> EnumerateFiles(string path, string searchPattern, SearchOption searchOption);
-    void CreateDirectory(string path);
-    void DeleteDirectory(string path, bool recursive = false);
+    FileOperationResult CreateDirectory(string path);
+    FileOperationResult DeleteDirectory(string path, bool recursive = false);
     bool DirectoryExists(string path);
     bool FileExists(string path);
-    void MoveFile(string sourceFileName, string destFileName, bool overwrite = false);
-    void DeleteFile(string path);
+    FileOperationResult MoveFile(string sourceFileName, string destFileName, bool overwrite = false);
+    FileOperationResult DeleteFile(string path);
     DateTimeOffset GetFileCreationTime(string path);
-    string ReadAllTextFromFile(string file);
+    string? ReadAllTextFromFile(string file);
     Stream OpenFileForReading(string path);
-    Stream CreateFileForWriting(string path);
+    Stream OpenFileForReading(string path,
+        bool useAsync,
+        FileMode fileMode = FileMode.Open,
+        FileAccess fileAccess = FileAccess.Read,
+        FileShare fileShare = FileShare.ReadWrite,
+        int bufferSize = 4096);
+    (FileOperationResult, Stream) CreateFileForWriting(string path);
```

Comment:
I wonder if it would be better to return the Stream as an out parameter here and have this method return a `FileOperationResult`.